### ISSUE TITLE
[Feat / Performance] Ignore updateTriggers on invisible layer

### ISSUE THREAD

**Issue by elfjes** [visgl/deck.gl#6836]
### Target Use Case

On layers with `visible: false`, a change in `updateTriggers` currently causes deck.gl to still call the associated accessor(s) again. This seems to be uneccesary, the accessors only need to be reevaluated once the layer becomes visible again. At the moment, naively changing the updateTriggers of invisible layers now causes a significant performance overhead, especially when the accessors are relatively expensive to run

### Proposal

On layers that become `visible: false`, store the last value of `updateTriggers` (from when the layer was `visible: true`), then only once the layer becomes `visible: true` again, evaluate any changes to `updateTriggers` and call the associated accessor(s)

**Comment by Pessimistress** [comment]
Just curious, why do you update the `updateTriggers` if you do not wish for the layer to be updated?

**Comment by elfjes** [comment]
Good question! Our accessors are `timestamp` dependent, so we add the current `timestamp` as an updateTrigger, which is done independently from setting the `visible` prop (when a user toggles a layer on or off). What we could do, is drop the `updateTriggers` entirely for layers that are `visible: false`. This works well, except that when a layer is toggled visible (and the `updateTriggers` are added to the layer again) this triggers an update to the accessors, even though they might not have changed. It's not a very big deal I guess, since a single additional update of the accessors is not very expensive, but still it seems unnecessary. 

A next step would be to keep track of the last `updateTriggers` prior to setting `visible: false`, and make sure to set the layer `updateTriggers` to that value as long as the layer is `visible: false` and only set any new values once the layer becomes `visible: true`, and I guess that more or less mimics the behaviour I'm proposing for deck.gl

**Comment by Pessimistress** [comment]
There have been similar feature requests before (generally block updates on invisible layers). I am hesitant to implement something simply based on `props.visible`, because layer visibility can be determined by a handful of factors, including `layerFilter` and `layer.parent.filterSubLayer`. Whether a layer is visible is context-dependent and not always inferable from within its own scope.

**Comment by Pessimistress** [comment]
Duplicate of #6056
